apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/FakeTracer.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright 2018 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using TraceProto = Google.Cloud.Trace.V1.Trace;
+
+namespace Google.Cloud.Diagnostics.Common.Tests
+{
+    /// <summary>
+    /// An <see cref="IConsumer{T}"> for testing that does nothing.
+    /// </summary>
+    public class FakeConsumer : IConsumer<TraceProto>
```

Comment:
*Possibly* rename to NoOpConsumer as later we may want a FakeConsumer that remembers all the traces it receives? Although it wouldn't hurt to add that functionality to this one instead...

Reply:
I'm renaming to NoOpConsumer. In real code there's no way for clients of IConsumer to access the traces it receives or any information about them, so I'm not sure why we would need a Fake to remember them? Unless we'd want to either verify something or expect some very specific behavior from the fake in which case we would use a Mock or write a specific Fake? Am I missing something?
If I'm wrong we can always add the remembering Fake when it's needed.

Reply:
We'd want a fake to remember them so that we could check what was passed to the consumer. I typically prefer fakes over mocks as they're less brittle in the face of irrelevant changes, but it would depend on the specific situation. I'm fine with it being NoOpConsumer for now, certainly :)